Add readme and an example (#7)
crossplane/oam-kubernetes-runtime: README.md

> # OAM Kubernetes Runtime

Reviewer: A Github badge just a link to https://pkg.go.dev/mod/github.com/crossplane/oam-kubernetes-runtime may help folks discover the API docs.

Author: what is a Github badge?

> # oam-kubernetes-runtime Project

Reviewer 2: ```suggestion
# OAM Kubernetes Runtime
```

> The OAM Kubernetes runtime project is a collection of Golang help libraries and official api type

Reviewer 3: ```suggestion
The OAM Kubernetes runtime project is a collection of Golang helper libraries and api
```

> The OAM Kubernetes runtime project is a collection of Golang help libraries and official api type

Reviewer 2: My suggestion:

The OAM Kubernetes Runtime is the building block of data model in [Crossplane](https://github.com/crossplane/crossplane). It's basically a collection of Golang help libraries and api types to make implementing [Open Application Model](https://github.com/oam-dev/spec) easier on Kubernetes. 

OAM Kubernetes Runtime is designed to be platform builder facing and embedded into a larger system, for end-users facing experience of OAM, please always check [Crossplane official documentation](https://crossplane.github.io/docs).

>  definitions for building an OAM Kubernetes runtime that is compatible with the latest stable

Reviewer 3: ```suggestion
 definitions for building an OAM Kubernetes runtime
```

>   OAM spec. 

Reviewer 3: ```suggestion
```

## Prerequisites

> 1. go version 1.12+

Reviewer 4: I believe we are using go `1.13` here, but I don't think this a strict prerequisite for consuming?

Author: I believe this is related to the go module usage.  Maybe go 1.11 also works but I haven't tried.

> 2. Kubernetes version v1.15+ with KubeCtl configured 

Reviewer 3: ```suggestion
2. Kubernetes version v1.15+ with kubectl configured 
```

## Getting started

The functionality of this library can be demonstrated with the following steps:

* Fork and Clone this project

* Build the library 

```shell
make submodules 

make generate

make
```

* Install CRDs to your Kubernetes cluster

```shell
kubectl apply -f cluster/charts/oam-kubernetes-runtime/crds/

> ```

Reviewer 4: These are being removed as it makes more sense for an implementation to generate their own CRDs

Author: in this case, I will add a make CRD command in the make file so that this example can be self-contained.

> ```

Reviewer: `make test` and `make reviewable` are also handy targets that may be worth documenting. The former runs the unit tests, while the latter runs linters, generation, etc to catch some of the things that might cause CI to fail.

Author: sure, we can add some instructions in the "how to contribute" section.

* Run OAM sample controller
```
go run pkg/examples/containerized-workload/main.go
```

* Apply the sample application config

```
kubectl apply -f pkg/examples/containerized-workload/sample_application_config.yaml
 
```

* Verify that corresponding CRs are emitted. 

You should see a containerizedworkloads looking like below
```
kubectl get containerizedworkloads.core.oam.dev  
NAME                         AGE
example-appconfig-workload   12s
```

And a manualscalertraits looking like below
```
kubectl get manualscalertraits.core.oam.dev
NAME                      AGE
example-appconfig-trait   54s
```

> ## Goals

Reviewer 2: Let's delete this section, I've add a suggestion to include goals at the beginning.

> # OAM Kubernetes Project

Reviewer 2: It's not common to say xxx project in general, especially for a building block library.

```suggestion
# OAM Kubernetes Runtime
```

The OAM Kubernetes runtime project is a collection of Golang helper libraries and OAM api type

>  definitions. 

Reviewer 4: ```suggestion
definitions. 
```

> It is designed to help OAM platform builders rather than being used directly by developers

Reviewer 2: These two paragraphs are verbose and repeating. 

Could you compress the main points in one paragraph (i.e. this repo is platform builder facing and important for ensuring interoperability in OAM platforms)?

Author: I actually expended a bit here to make our point loud and clear.

Reviewer 2: That should be the reason I feel it's verbose :-) Personally I'd prefer one sentence/paragraph to tell out the main points.

Reviewer: What is the difference between an "OAM platform builder" and a "developer"? Does developer mean "folks developing OAM", or is it synonymous with "end-user" in this context?

Author: a developer is an OAM application developer, aka "end-user".  Do you prefer to the term "application developer" or "end-user"?

or end-users. We would recommend end-users to check
[Crossplane  official  documentation](https://crossplane.github.io/docs) instead.

In addition, we created this library with the following goals in mind  
* All OAM Kubernetes platform builders use and contribute to this library. 
* The wide adoption of this library can facilitate workload/trait interoperability among OAM
 Kubernetes platform builders.
 
## Prerequisites

1. Golang version 1.12+
2. Kubernetes version v1.15+ with kubectl configured

## Getting started

The functionality of this library can be demonstrated with the following steps:

* Fork and Clone this project

* Build the library 

```shell
make submodules 

make
```

* Generate and install CRDs to your Kubernetes cluster

```shell
make generate

kubectl apply -f crds/
```

* Run OAM sample controller
```
go run examples/containerized-workload/main.go
```

* Apply the sample application configurations

```
kubectl apply -f examples/containerized-workload/ 
```

* Verify that corresponding CRs are emitted. 

> You should see a containerizedworkloads looking like below

Reviewer 4: ```suggestion
You should see a `ContainerizedWorkload` looking like below
```

```
kubectl get containerizedworkloads.core.oam.dev  
NAME                         AGE
example-appconfig-workload   12s
```

> And a manualscalertraits looking like below

Reviewer 4: ```suggestion
And a `ManualScalerTrait` looking like below
```
